functions/groups.js

```diff
@@ -50,7 +50,7 @@ const AssignGroups = {
   usesContext: true,
   mutations: ['persons', 'schedule'],
   implementation: (ctx, round, assignmentSets, scorers, stationRules, attemptNumber, overwrite) => {
-    return assign.Assign(ctx.competition, round, assignmentSets, scorers, stationRules, attemptNumber, overwrite || ctx.dryrun)
+    return assign.Assign(ctx.competition, round, assignmentSets, scorers, stationRules, attemptNumber, overwrite)
```

Comment:
Hmm, this one was intentional -- the goal was to make it easier to iterate on groups in dry-run mode, while using the same commands you would use to generate groups normally. Adding overwrite=true is easy enough though, and I can see the argument for "dry-run should behave identically to non-dry-run, except for the write".

I'm fine going ahead with this, but I might revisit when I start working on NAC groups if it makes things harder then. I'm curious what was the reason you made this change?

Reply:
I can absolutely explain why I had to do this: I did a non-dry-run to actually assign some groups, without setting overwrite.
Later I ran the same script in dry-run mode, and since I did not set overwrite I assumed it would just skip these assignments, but it did not! And then I realized that because I was on dry-run mode it would automatically assign anyway.

I must admit that I did not investigate a lot the groups I generated, and therefore I did not have to do this back and forth "generate groups for real", then "adjust them in dry run mode".
I'm definitely fine with setting overwrite to true manually though, when it's needed to ignore what is saved on the WCA website!

But generally I can also adjust the way I'm working with compscript; I'm still trying to keep a workflow where I can re-run the full script and where it doesn't mess up what is in there already, but does create what is missing if needed.

Reply:
I'm convinced :)